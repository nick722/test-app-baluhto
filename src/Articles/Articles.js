import * as React from 'react';

const MyListItem = (props) => (
  <li key={props.id}>
    <button onClick={props.clickHandler}>{props.children}</button>
  </li>
);

// TODO:
// - Loading indicator
// - Render Articles
// - Add sponsor badge

class Articles extends React.Component {

  componentDidMount() {
    this.props.actions.loadArticles();
  }

  clickHandler = () => {
    console.log('CLICK!!!');
    this.props.clickHandler();
  }

  renderArticles = (articles = []) => {
    console.log('ARTICLES >>> ', articles);
    return articles.map((article) => (
      <MyListItem clickHandler={this.clickHandler} id={article.id}>{article.fields.title}</MyListItem>
    ))
  }

  render() {
    return (
      <ul>
        {this.renderArticles(this.props.articles.data)}
      </ul>
    );
  }
}

export default Articles;
