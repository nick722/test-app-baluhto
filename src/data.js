export const articles = {
  data: [
    {
      id: 'article0',
      fields: {
        title: 'Article 0',
        author: {
          id: 'author0',
          name: 'Author 0',
        },
        content: 'Article 0 content',
      },
    },
    {
      id: 'article1',
      fields: {
        title: 'Article 1',
        author: {
          id: 'author0',
          name: 'Author 0',
        },
        content: 'Article 1 content',
      },
    },
    {
      id: 'article2',
      fields: {
        title: 'Article 2',
        author: {
          id: 'author1',
          name: 'Author 1',
        },
        content: 'Article 2 content',
      },
    },
  ],
}
