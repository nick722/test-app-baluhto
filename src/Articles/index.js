import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import Articles from './Articles';
import * as actions from './reducer/actions'
import reducer from './reducer/articles';

const mapStateToProps = (state) => ({
  articles: state.articles,
});

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators(actions, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(Articles);
export { reducer };
