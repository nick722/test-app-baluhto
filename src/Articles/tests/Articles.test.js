import * as React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Articles from '../Articles';
import * as data from '../../data';

configure({ adapter: new Adapter() });

describe('Articles', () => {
  const setup = (props = {}) => shallow(
    <Articles
      articles={{ data: [] }}
      actions={{
        loadArticles: jest.fn(),
      }}
      {...props}
    />
  );

  it('should match snapshot', () => {
    const wrapper = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render articles', () => {
    const articles = { data: data.articles.data }
    const wrapper = setup({ articles });

    expect(wrapper.find('MyListItem')).toHaveLength(data.articles.data.length);
  });

  it('should call clickHandler', () => {
    const articles = { data: data.articles.data };
    const clickHandler = jest.fn();
    const wrapper = setup({ articles, clickHandler });
    const first = wrapper.find('MyListItem').first();
    first.props().clickHandler();

    wrapper.setState();
  });

  it('should call clickHandler method', () => {
    const articles = { data: data.articles.data };
    const clickHandler = jest.fn();
    const wrapper = setup({ articles, clickHandler });
    wrapper.instance().clickHandler();
    expect(wrapper.state('myKey')).toBe({});

    
  });
});