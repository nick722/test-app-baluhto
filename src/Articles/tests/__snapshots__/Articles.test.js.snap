// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles should match snapshot 1`] = `<ul />`;
